src/Core/Order/Repository/OrderRepositoryInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Order\Repository;
+
+use Order;
+use PrestaShop\PrestaShop\Core\Domain\Order\ValueObject\OrderId;
+
+interface OrderRepositoryInterface
+{
+    public function get(OrderId $orderId): Order;
```

Comment:
I think we can get rid of this legacy class in the core, we won't be able to remove it afterwards.
Order should have it's own interface also to make it compatible if possible. You can find an example here : https://github.com/Sylius/Order/blob/master/Model/OrderInterface.php

```suggestion
    public function get(OrderId $orderId);
```

Reply:
All the other repositories return legacy class as well, won't this mean that once we replace object model we will need to change them all anyway?

Reply:
No, because if you are using an interface, you'll be able to construct a "new" object, with the methods. If you return legacy classes, we'll stick with public properties, magic getters, etc. We don't want to do that here.

Reply:
As I commented in another comment, this is not the moment to introduce such an interface, especially on the current OrderRepository which is meant to handle ObjectModel not generic interfaces

The work on generic core model interfaces and associated repositories will be done later but it needs some thought before, so you should remove this interface for now as it's not adapted